gcloud-java-dns/src/test/java/com/google/gcloud/dns/DnsBatchTest.java

```diff
@@ -312,10 +340,211 @@ public void testGetProjectWithOptions() {
     String selector = (String) capturedOptions.getValue().get(PROJECT_FIELDS.rpcOption());
     assertTrue(selector.contains(Dns.ProjectField.QUOTA.selector()));
     assertTrue(selector.contains(Dns.ProjectField.PROJECT_ID.selector()));
-    DnsRpc.Callback<Project> capturedCallback = callback.getValue();
+    RpcBatch.Callback<Project> capturedCallback = callback.getValue();
     capturedCallback.onSuccess(PROJECT_INFO.toPb());
     assertEquals(PROJECT_INFO, batchResult.get());
   }
 
-  // todo(mderka) test submit and other methods when implemented
+  @Test
+  public void testListRecordSets() {
+    EasyMock.reset(batchMock);
+    Capture<RpcBatch.Callback<ResourceRecordSetsListResponse>> callback = Capture.newInstance();
+    Capture<Map<DnsRpc.Option, Object>> capturedOptions = Capture.newInstance();
+    batchMock.addListRecordSets(EasyMock.eq(ZONE_NAME), EasyMock.capture(callback),
+        EasyMock.capture(capturedOptions));
+    EasyMock.replay(batchMock);
+    DnsBatchResult<Page<RecordSet>> batchResult = dnsBatch.listRecordSets(ZONE_NAME);
+    assertEquals(0, capturedOptions.getValue().size());
+    assertNotNull(callback.getValue());
```

Comment:
I believe here we should find a way to check that the callback is correctly created and will correctly forward success and failure. Do you think something like the following would make sense:
```java
RpcBatch.Callback<ResourceRecordSetsListResponse> capturedCallback = callback.getValue();
ResourceRecordSetsListResponse response = new ResourceRecordSetsListResponse()
    .setRrsets(SOME_RECORD_SETS_LIST)
    .setNextPageToken(SOME_PAGE_TOKEN);
capturedCallback.onSuccess(response);
// here we check that the previously returned DnsBatchResult has been correctly populated
Page<RecordSet> page = batchResult.get();
assertEquals(SOME_PAGE_TOKEN, page.nextPageCursor());
// here compare the two returned lists
```
same deal for errors:
```java
GoogleJsonError error = new GoogleJsonError()
    .setCode(SOME_CODE)
    .setMessage(SOME_MESSAGE);
capturedCallback.onFailure(error);
// here we check that the previously returned DnsBatchResult has been correctly populated
try {
    batchResult.get();
    fail("Should throw a DnsException on error.");
} catch (DnsException ex) {
    // expected
    assertEquals(SOME_CODE, ex.code());
    assertEquals(SOME_MESSAGE, ex.getMessage());
}
```

I might be totally off, what do you think?